controllers/batchscheduler/factory.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package batchscheduler
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/klog"
+
+	schedulerinterface "github.com/googlecloudplatform/flink-operator/controllers/batchscheduler/interface"
+	"github.com/googlecloudplatform/flink-operator/controllers/batchscheduler/volcano"
+)
+
+var (
+	mutex            sync.Mutex
+	once             = sync.Once{}
+	schedulerPlugins = map[string]schedulerinterface.BatchScheduler{}
+)
+
+func lazyInitialize() {
```

Comment:
How long does the init usually take? Each iteration of the reconciliation loop runs sequentially, is it really necessary to have lazy init and concurrency control for this module?

Reply:
it does not need network connection. So it should not be a matter

Reply:
Then can we remove the lazy init and mutex? let's keep it simple.

Reply:
Will do that